packages/plus/src/components/EstimateCost/EstimateCostProvider.tsx

```diff
@@ -0,0 +1,68 @@
+import deepMerge from 'deepmerge'
+import type { ReactNode } from 'react'
+import { createContext, useCallback, useContext, useMemo } from 'react'
+import EstimateCostLocales from './locales/en'
+
+const EstimateCostContext = createContext<{
+  locales: typeof EstimateCostLocales
+  formatNumber: (number: number, options: FormatNumberOption) => string
+}>({ locales: EstimateCostLocales, formatNumber: () => '' })
+
+export const useEstimateCost = () => useContext(EstimateCostContext)
+
+type EstimateCostProviderProps = {
+  children: ReactNode
+  locales?: typeof EstimateCostLocales
+  currency?: string
+  numberLocales?: string
+}
+
+type FormatNumberOption = {
+  currencyDisplay?: 'symbol' | 'code' | 'name'
+  unit?: string
+  unitDisplay?: 'long' | 'short' | 'narrow'
+  useGrouping?: boolean
+  minimumIntegerDigits?: number
+  minimumFractionDigits?: number
+  maximumFractionDigits?: number
+  minimumSignificantDigits?: number
+  maximumSignificantDigits?: number
+}
+
+export const EstimateCostProvider = ({
+  children,
+  locales,
+  currency,
+  numberLocales,
+}: EstimateCostProviderProps) => {
+  const newLocales = locales
+    ? deepMerge(EstimateCostLocales, locales)
+    : EstimateCostLocales
+
+  const formatNumber = useCallback(
```

Comment:
this function is a duplicate of what we have into scaleway/use-i18n right ?

Reply:
@philibea not same but different but still the same..

Yeah it's a bit different as it focus on currency only and I cannot import the one from i18n as it needs a provider so I made home made one for this component but nothing crazy I only use Intl.NumberFormat at it's most raw usage.